storage/bootstrap/types.go

```diff
@@ -50,34 +72,169 @@ type ShardResult interface {
 	// RemoveSeries removes a single series of blocks.
 	RemoveSeries(id string)
 
-	// GetAllSeries returns all series of blocks.
-	GetAllSeries() map[string]block.DatabaseSeriesBlocks
-
 	// Close closes a shard result.
 	Close()
 }
 
+// ShardResults is a map of shards to shard results.
+type ShardResults map[uint32]ShardResult
+
+// AddResults adds other shard results to the current shard results.
+func (r ShardResults) AddResults(other ShardResults) {
+	for shard, result := range other {
+		if existing, ok := r[shard]; ok {
+			existing.AddResult(result)
+		} else {
+			r[shard] = result
+		}
+	}
+}
+
+// ShardTimeRanges is a map of shards to time ranges.
+type ShardTimeRanges map[uint32]xtime.Ranges
+
+// IsEmpty returns whether the shard time ranges is empty or not.
+func (r ShardTimeRanges) IsEmpty() bool {
+	for _, ranges := range r {
+		if !xtime.IsEmpty(ranges) {
+			return false
+		}
+	}
+	return true
+}
+
+// Equal returns whether two shard time ranges are equal.
+func (r ShardTimeRanges) Equal(other ShardTimeRanges) bool {
+	if len(r) != len(other) {
+		return false
+	}
+	for shard, ranges := range r {
+		otherRanges, ok := other[shard]
+		if !ok {
+			return false
+		}
+		matched := 0
+		it := ranges.Iter()
+		otherIt := otherRanges.Iter()
+		if it.Next() && otherIt.Next() {
+			value := it.Value()
+			otherValue := otherIt.Value()
+			if !value.Start.Equal(otherValue.Start) ||
+				!value.End.Equal(otherValue.End) {
+				return false
+			}
+			matched++
+		}
+		if matched != ranges.Len() {
```

Comment:
This is in case it.Next() returns false even though it said it had a certain length.  Thoughts on keeping this check or discarding?  It does serve some purpose, but then I don't know how likely it is that it could ever happen.